### PR TITLE
r/synthetics_canary - fix canary update

### LINKED ISSUE #17642: Synthetics Canary update not working
### Terraform CLI and Terraform AWS Provider Version

Terraform v0.14.6
+ provider registry.terraform.io/hashicorp/archive v2.0.0
+ provider registry.terraform.io/hashicorp/aws v3.28.0
+ provider registry.terraform.io/hashicorp/null v3.0.0

### Affected Resource(s)

aws_synthetics_canary
### Terraform Configuration Files

```
  ~ resource "aws_synthetics_canary" "synthetics" {
        id                       = "cm-dev-testui"
        name                     = "cm-dev-testui"
      ~ zip_file                 = "files/testui-canary-298c366a80006e33f3b97d2dbc9a4c8f.zip" -> "files/testui-canary-082d98273331513b8d3c4ae575c3fbb2.zip"
        # (12 unchanged attributes hidden)


        # (2 unchanged blocks hidden)
    }
```

### Debug Output

2021-02-16T13:27:43.115-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:27:43 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/GetCanary Details:

State is RUNNING

2021-02-16T13:27:47.372-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:27:47 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/StopCanary Details:

2021-02-16T13:27:47.590-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:27:47 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/GetCanary Details:

State is STOPPING

2021-02-16T13:27:48.610-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:27:48 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/GetCanary Details:

State is STOPPED

2021-02-16T13:27:48.795-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:27:48 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/UpdateCanary Details:

State is UPDATING

2021-02-16T13:28:13.377-0500 [INFO]  plugin.terraform-provider-aws_v3.28.0_x5: 2021/02/16 13:28:13 [DEBUG] [aws-sdk-go] DEBUG: Request synthetics/GetCanary Details:

State is STOPPED

2021/02/16 13:28:13 [DEBUG] aws_synthetics_canary.synthetics[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: error waiting for Synthetics Canary (cm-dev-testui) updating: unexpected state 'STOPPED', wanted target 'READY'. last error: %!s(<nil>)

Error: error waiting for Synthetics Canary (cm-dev-testui) updating: unexpected state 'STOPPED', wanted target 'READY'. last error: %!s(<nil>)

### Expected Behavior

canary should have been restarted but instead terraform update fails.

### Actual Behavior

looks like the code is expecting the canary to be in READY state after update but CloudWatch Synthetics is putting the canary backed to STOPPED state since it was in STOPPED state before UpdateCanary API was called. It is likely that AWS updated behavior after the developer had tested updating canary scenario.

### Steps to Reproduce

Update the canary code and run terraform apply

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #17642

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSyntheticsCanary_'
--- PASS: TestAccAWSSyntheticsCanary_disappears (102.15s)
--- PASS: TestAccAWSSyntheticsCanary_s3 (114.20s)
--- PASS: TestAccAWSSyntheticsCanary_basic (164.13s)
--- PASS: TestAccAWSSyntheticsCanary_runtimeVersion (172.09s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary_codeChanges (175.97s)
--- PASS: TestAccAWSSyntheticsCanary_tags (206.14s)
--- PASS: TestAccAWSSyntheticsCanary_runConfig (211.43s)
--- PASS: TestAccAWSSyntheticsCanary_runConfigTracing (217.31s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary (228.48s)
--- PASS: TestAccAWSSyntheticsCanary_vpc (778.07s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSyntheticsCanary_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSyntheticsCanary_ -timeout 120m
=== RUN   TestAccAWSSyntheticsCanary_basic
=== PAUSE TestAccAWSSyntheticsCanary_basic
=== RUN   TestAccAWSSyntheticsCanary_runtimeVersion
=== PAUSE TestAccAWSSyntheticsCanary_runtimeVersion
=== RUN   TestAccAWSSyntheticsCanary_startCanary
=== PAUSE TestAccAWSSyntheticsCanary_startCanary
=== RUN   TestAccAWSSyntheticsCanary_startCanary_codeChanges
=== PAUSE TestAccAWSSyntheticsCanary_startCanary_codeChanges
=== RUN   TestAccAWSSyntheticsCanary_s3
=== PAUSE TestAccAWSSyntheticsCanary_s3
=== RUN   TestAccAWSSyntheticsCanary_runConfig
=== PAUSE TestAccAWSSyntheticsCanary_runConfig
=== RUN   TestAccAWSSyntheticsCanary_runConfigTracing
=== PAUSE TestAccAWSSyntheticsCanary_runConfigTracing
=== RUN   TestAccAWSSyntheticsCanary_vpc
=== PAUSE TestAccAWSSyntheticsCanary_vpc
=== RUN   TestAccAWSSyntheticsCanary_tags
=== PAUSE TestAccAWSSyntheticsCanary_tags
=== RUN   TestAccAWSSyntheticsCanary_disappears
=== PAUSE TestAccAWSSyntheticsCanary_disappears
=== CONT  TestAccAWSSyntheticsCanary_basic
=== CONT  TestAccAWSSyntheticsCanary_disappears
--- PASS: TestAccAWSSyntheticsCanary_disappears (44.09s)
=== CONT  TestAccAWSSyntheticsCanary_tags
--- PASS: TestAccAWSSyntheticsCanary_basic (94.56s)
=== CONT  TestAccAWSSyntheticsCanary_vpc
--- PASS: TestAccAWSSyntheticsCanary_tags (87.27s)
=== CONT  TestAccAWSSyntheticsCanary_runConfigTracing
=== CONT  TestAccAWSSyntheticsCanary_runConfig
--- PASS: TestAccAWSSyntheticsCanary_runConfigTracing (87.67s)
--- PASS: TestAccAWSSyntheticsCanary_runConfig (89.22s)
=== CONT  TestAccAWSSyntheticsCanary_s3
--- PASS: TestAccAWSSyntheticsCanary_s3 (59.08s)
=== CONT  TestAccAWSSyntheticsCanary_startCanary_codeChanges
=== CONT  TestAccAWSSyntheticsCanary_startCanary
--- PASS: TestAccAWSSyntheticsCanary_startCanary_codeChanges (86.36s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary (91.85s)
=== CONT  TestAccAWSSyntheticsCanary_runtimeVersion
--- PASS: TestAccAWSSyntheticsCanary_runtimeVersion (81.95s)
--- PASS: TestAccAWSSyntheticsCanary_vpc (666.49s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	761.131s
```

File: .changelog/17704.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (bflad) [inline .changelog/17704.txt:1]: This seems like a bug fix. 😉 

``````suggestion
```release-note:bug
`````

**Review by bflad — APPROVED** [review]
One minor thing, otherwise looks great, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSSyntheticsCanary_disappears (59.10s)
--- PASS: TestAccAWSSyntheticsCanary_s3 (65.68s)
--- PASS: TestAccAWSSyntheticsCanary_basic (92.10s)
--- PASS: TestAccAWSSyntheticsCanary_runConfigTracing (93.37s)
--- PASS: TestAccAWSSyntheticsCanary_runConfig (93.68s)
--- PASS: TestAccAWSSyntheticsCanary_tags (95.60s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary_codeChanges (104.92s)
--- PASS: TestAccAWSSyntheticsCanary_runtimeVersion (105.96s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary (117.58s)
--- PASS: TestAccAWSSyntheticsCanary_vpc (693.51s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSSyntheticsCanary_disappears (63.23s)
--- PASS: TestAccAWSSyntheticsCanary_s3 (67.62s)
--- PASS: TestAccAWSSyntheticsCanary_basic (84.47s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary_codeChanges (85.49s)
--- PASS: TestAccAWSSyntheticsCanary_runtimeVersion (111.19s)
--- PASS: TestAccAWSSyntheticsCanary_tags (111.24s)
--- PASS: TestAccAWSSyntheticsCanary_runConfigTracing (112.80s)
--- PASS: TestAccAWSSyntheticsCanary_runConfig (112.90s)
--- PASS: TestAccAWSSyntheticsCanary_startCanary (114.62s)
--- PASS: TestAccAWSSyntheticsCanary_vpc (1954.15s)
```

File: .changelog/17704.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+`````
```
Comment (bflad) [inline .changelog/17704.txt:2]:
```suggestion
```

**Pull request merged**